### DISCUSSION TITLE
Possibility to use as not only read-only gateway

### DISCUSSION THREAD

**Discussion by andrew-demb** [nginxinc/nginx-s3-gateway#114] [Q&A]
Thanks for the package.

We want to use all current features of `nginx-s3-gateway`, but also to use it to be a proxy for mutation requests (put file, delete file, etc) from the internal network from applications too.

We used `minio` before, but they deprecated gateway mode support and we currently searching alternative solution.

Necessary features for us for mutation requests: 
- hide from the application real credentials to the cloud S3, cloud provider
- don't initiate connections to the external cloud S3 from the stateless application (PHP)

As we found in the codebase [1][2] there is no support at the current moment.

Is it possible to add support for mutating requests in `nginx-s3-gateway` or it is totally out of scope?

---

Thanks in advance.

---

[1] https://github.com/nginxinc/nginx-s3-gateway/blob/e3068dfbef7619000f7cb06ff5f251022d72d967/common/etc/nginx/include/s3gateway.js#L344-L350
[2] https://github.com/nginxinc/nginx-s3-gateway/blob/e3068dfbef7619000f7cb06ff5f251022d72d967/common/docker-entrypoint.sh#L54-L60C3

**Comment by dekobon** [comment]
Hi @andrew-demb , it is something that we have wanted to add for a while. It is just a matter of doing the work and teasing out the issues.

**Reply by dekobon** [reply]
Note to self or other future implementors: the feature we will probably need to use to accomplish this with v4 AWS auth signatures is [Unsigned-Payload](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html).
